File: os-hub/internal/models/db/device.go
package db

import (
	"time"
)

type Device struct {
	ID           uint      `gorm:"primaryKey;autoIncrement"`
	MacAddress   string    `gorm:"not null;uniqueIndex"`
	IPAddress    string    `gorm:"not null"`
	Port         int       `gorm:"not null"`
	Name         string    `gorm:"not null"`
	Description  *string   `gorm:"type:text"`
	ServiceType  *string   `gorm:"type:text"` // mDNS service type, e.g., "_http._tcp", will be null if not discovered via mDNS
	DeviceType   string    `gorm:"not null"`
	InstanceType string    `gorm:"not null"`
	InstanceName string    `gorm:"not null"`
	LastSeen     time.Time `gorm:"autoCreateTime"`
	CreatedAt    time.Time `gorm:"autoCreateTime"`
	UpdatedAt    time.Time `gorm:"autoUpdateTime"`
}

File: os-management/src/domain/common/device.ts
export type DeviceType = "entry" | "other";

export interface BaseDeviceData {
  id: number;
  macAddress: string;
  ipAddress: string;
  port: number;
  deviceType: DeviceType;
  instanceType: string;
  instanceName?: string;
  name?: string;
  description?: string;
  isOnline: boolean;
  lastSeen: number;
  createdAt: number;
  updatedAt: number;
}

export interface EntryDeviceData extends BaseDeviceData {
  deviceType: "entry";
  lockStatus: "LOCKED" | "UNLOCKED" | "UNKNOWN";
}
